lib/index.js

```diff
@@ -0,0 +1,243 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const base64url = require('base64url-universal');
+const bedrock = require('bedrock');
+const brPackageManager = require('bedrock-package-manager');
+const crypto = require('crypto');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const {util: {BedrockError}} = bedrock;
+
+// load config defaults
+require('./config');
+
+const AES_KW_ALGORITHM = 'id-aes256-wrap';
+const AES_KW_RFC3394_IV = Buffer.from('A6A6A6A6A6A6A6A6', 'hex');
+
+bedrock.events.on('bedrock.init', () => brPackageManager.register({
+  alias: 'ssm-v1',
+  packageName: 'bedrock-ssm-mongodb',
+  type: 'bedrock-kms-store',
+}));
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await promisify(database.openCollections)(['ssm']);
+
+  await promisify(database.createIndexes)([{
+    // cover queries by ID
+    collection: 'ssm',
+    fields: {id: 1},
+    options: {unique: true, background: false}
+  }]);
+});
+
+/**
+ * @module bedrock-ssm-mongodb
+ */
+
+/**
+ * Generates a new key.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.keyId - The key ID to use.
+ * @param {Object} options.operation - The KMS operation.
+ *
+ * @returns {Promise<Object>} An object containing `{id}`.
+ */
+exports.generateKey = async ({keyId, operation}) => {
+  assert.string(keyId, 'options.keyId');
+  assert.object(operation, 'options.operation');
+
+  let key;
+
+  const {invocationTarget: {type}} = operation;
+  if(type === 'AesKeyWrappingKey2019') {
+    // TODO: support other lengths?
+    key = {
+      type,
+      secret: base64url.encode(crypto.randomBytes(32))
+    };
+  } else if(type === 'Sha256HmacKey2019') {
+    // TODO: support other hashes?
+    key = {
+      type,
```

Comment:
The `keyId` is not being stored in the body, is that a feature or a bug?

Reply:
I think it started out as a feature and now it's a bug. So ... it's a bug.

Reply:
Fixed.